awswrangler/dynamodb/_read.py

```diff
@@ -146,3 +150,331 @@ def read_partiql_query(
+        Boto3 Session. Defaults to None (the default boto3 Session will be used).
+
+    Returns
+    -------
+    Sequence[Mapping[str, Any]]
+        Retrieved items.
+    """
+    # Get DynamoDB resource and Table instance
+    resource = _utils.resource(service_name="dynamodb", session=boto3_session)
+    table = get_table(table_name=table_name, boto3_session=boto3_session)
+
+    # Extract 'Keys' from provided kwargs: if needed, will be reinserted later on
+    keys = kwargs.pop("Keys", None)
+
+    # Conditionally define optimal reading strategy
+    use_get_item = (keys is not None) and (len(keys) == 1)
+    use_batch_get_item = (keys is not None) and (len(keys) > 1)
+    use_query = (keys is None) and ("KeyConditionExpression" in kwargs)
+    use_scan = (keys is None) and ("KeyConditionExpression" not in kwargs)
+
+    # Read items
+    if use_get_item:
+        kwargs["Key"] = keys[0]
+        items = [table.get_item(**kwargs).get("Item", {})]
+    elif use_batch_get_item:
+        kwargs["Keys"] = keys
+        response = resource.batch_get_item(RequestItems={table_name: kwargs})
+        items = response.get("Responses", {table_name: []}).get(table_name, [])
+        # SEE: handle possible unprocessed keys. As suggested in Boto3 docs,
+        # this approach should involve exponential backoff, but this should be
+        # already managed by AWS SDK itself, as stated
+        # [here](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Programming.Errors.html)
+        while response["UnprocessedKeys"]:
+            kwargs["Keys"] = response["UnprocessedKeys"][table_name]["Keys"]
+            response = resource.batch_get_item(RequestItems={table_name: kwargs})
+            items.extend(response.get("Responses", {table_name: []}).get(table_name, []))
+    elif use_query or use_scan:
+        _read_method = table.query if use_query else table.scan
+        response = _read_method(**kwargs)
+        items = response.get("Items", [])
+
+        # Handle pagination
+        while "LastEvaluatedKey" in response:
+            kwargs["ExclusiveStartKey"] = response["LastEvaluatedKey"]
+            response = _read_method(**kwargs)
+            items.extend(response.get("Items", []))
+
+    return items
+
+
+@apply_configs
+def read_items(
+    table_name: str,
+    partition_values: Optional[Sequence[Any]] = None,
+    sort_values: Optional[Sequence[Any]] = None,
+    filter_expression: Optional[Union[ConditionBase, str]] = None,
+    key_condition_expression: Optional[Union[ConditionBase, str]] = None,
+    expression_attribute_names: Optional[Dict[str, str]] = None,
+    expression_attribute_values: Optional[Dict[str, str]] = None,
+    consistent: bool = False,
+    columns: Optional[Sequence[str]] = None,
+    allow_full_scan: bool = False,
+    max_items_evaluated: Optional[int] = None,
+    as_dataframe: bool = True,
+    boto3_session: Optional[boto3.Session] = None,
+) -> Union[pd.DataFrame, List[Dict[str, Any]]]:
```

Comment:
We usually only return a pandas DataFrame given that this is supposed to be an `AWS SDK for pandas`. However I do see the appeal of also enabling list of dicts, especially that the `as_dataframe` argument is True by default

Reply:
Yeah, I totally understand the focus of awswrangler, but I thought that provide a pandas _opt-out_ choice would encourage the adoption of this new feature and related refactoring of pre-existing code.

Users can adopt it step-by-step: first by substituting their current snippets - which might return "raw" items as list of dicts - with `wr.dynamodb.read_items(..., as_dataframe=False)`, and only then dealing with a change in data structures.